new-cep.md

```diff
@@ -0,0 +1,204 @@
+- `on_exception`: runs when an exception is raised while a conda command is invoked
+
+Below, we discuss an example showing the use of the `pre_command` and `post_command` hooks together
+and an example of the `on_exception` hook.
+
+### `pre_command` and `post_command`
+
+Plugin authors create the `pre_command` and `post_command` hooks by first defining a `conda.plugins.hookimpl`
+decorated function called either `conda_pre_command` or `conda_post_command` which return either a `CondaPreCommand` 
+or `CondaPostCommand` class, respectively. These classes will both receive the following three properties:
+
+- `name`: unique name which identifies this plugin hook
+- `action`: a callable which contains the code to be run
+- `run_for`: a Python `set` of strings representing the commands this will be run on (e.g. `install` and `create`)
+
+#### Example
+
+```python
+from conda.plugins import hookimpl, CondaPreCommand, CondaPostCommand
+
+
+PLUGIN_NAME = "custom_plugin"
+
+
+def custom_plugin_pre_command_action():
+    """
+    Defines our custom pre-run action which simply prints a message.
+    """
+    print("pre-run action")
+
+
+@hookimpl
+def conda_pre_command():
+    """
+    Returns our CondaPreCommand class which attaches our ``custom_plugin_pre_command_action``
+    to the "install" and "create" command.
+    """
+    yield CondaPreCommand(
+        name=f"{PLUGIN_NAME}_pre_command",
+        action=custom_plugin_pre_command_action,
+        run_for={"install", "create"}
+    )
+
+
+def custom_plugin_post_command_action():
+    """
+    Defines our custom post-run action which simply prints a message.
+    """
+    print("post-run action")
+
+
+@hookimpl
+def conda_post_command():
+    """
+    Returns our CondaPreCommand class which attaches our ``custom_plugin_post_command_action`` to
+    the "install" and "create" command.
+    """
+    yield CondaPostCommand(
+        name=f"{PLUGIN_NAME}_post_command",
+        action=custom_plugin_post_command_action,
+        run_for={"install", "create"}
+    )
+```
+
+### `on_exception`
+
+For the `on_exception` hook, plugin authors begin by defining a `conda.plugins.hookimpl` decorated 
+function called `conda_on_exception`. This function will return the `CondaOnException` class with the
+following attributes:
+
+- `name`: unique name which identifies this plugin hook
+- `action`: a callable which contains the code to be run
+
+#### Example
+
+```python
+from conda.plugins import hookimpl, CondaOnException
+
+
+PLUGIN_NAME = "custom_plugin"
+
+
+def custom_plugin_on_exception_action():
+    """
+    Defines our custom on_exception action which simply prints a message.
+    """
+    print("on_exception action")
+
+
+@hookimpl
+def conda_on_exception():
+    """
+    Returns our CondaOnException class which attaches our ``custom_plugin_on_exception_action``.
+    """
+    yield CondaOnException(
+        name=f"{PLUGIN_NAME}_on_exception",
+        action=custom_plugin_on_exception_action
+    )
```

Comment:
It might make sense to make sure that the hook spec is set up as [`firstresult=True`](https://pluggy.readthedocs.io/en/stable/#first-result-only) to make sure we can define exception handlers that are final, and those that just inspect the exception and then move to the next registered exception handler.

Secondly, the hook implementation should receive the exception object as an argument and store it in the `CondaExceptionHandler` instance for later use. AND the actual handler callable should _also_ be passed the exception object so that it can do something with the exception.

Lastly, I think the naming is a little off about this, the hook yields handlers, so we should name the domain object `CondaExceptionHandler` and the hook spec `conda_exception_handlers`. 

```suggestion
from conda.plugins import hookimpl, CondaOnException


PLUGIN_NAME = "custom_plugin"


def print_exception(exception):
    """
    Defines our custom on_exception handler which prints a message
    with the provided exception object.
    """
    print(f"Caught exception: {exception}")


@hookimpl
def conda_exception_handlers(exception):
    """
    Returns our CondaExceptionHandler class which attaches our ``print_exception(``.
    """
    yield CondaExceptionHandler(
        name=f"{PLUGIN_NAME}_exception_handler",
        handler=print_exception,
        exception=exception,
    )
```